src/ToggleButton.js

```diff
@@ -44,42 +55,39 @@ const propTypes = {
    * @type {ReactRef}
    */
   inputRef: PropTypes.any,
-
-  /** @ignore */
-  innerRef: PropTypes.any,
 };
 
-class ToggleButton extends React.Component {
-  state = { focused: false };
-
-  handleFocus = e => {
-    if (e.target.tagName === 'INPUT') this.setState({ focused: true });
-  };
-
-  handleBlur = e => {
-    if (e.target.tagName === 'INPUT') this.setState({ focused: false });
-  };
-
-  render() {
-    const {
+const ToggleButton = React.forwardRef(
+  (
+    {
       children,
       name,
       className,
       checked,
       type,
       onChange,
+      toggleFocus,
       value,
       disabled,
+      focused,
```

Comment:
this should be state, or is there a larger plan that i'm not seeing here?

Reply:
there's no reason to control focused state. add a `[focus, setFocus] = useState(...)`